File: fitplanner-app/src/app/modules/user/interface/workout-plan.ts
import { UserCardioExercise } from "./user-cardio-exercise";
import { UserStrengthExercise } from "./user-strength-exercise";

export interface WorkoutPlan {
    date: string,
    strengthExerciseList: UserStrengthExercise[],
    cardioExerciseList: UserCardioExercise[]
}

File: fitplanner-services/authentication-service/src/main/java/com/fitplanner/authentication/model/training/WorkoutPlan.java
package com.fitplanner.authentication.model.training;

import java.util.List;

public class WorkoutPlan {

    private String date;
    private List<UserStrengthExercise> strengthExerciseList;
    private List<UserCardioExercise> cardioExerciseList;

    public WorkoutPlan() {}

    // getters
    public String getDate() { return date; }
    public List<UserStrengthExercise> getStrengthExerciseList() { return strengthExerciseList; }
    public List<UserCardioExercise> getCardioExerciseList() { return cardioExerciseList; }
}
